text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,1119 @@
+
+### To keep compatibility 
+
+We propose:
+- Deprecate the `platform.os` field from [package.toml](https://buildpacks.io/docs/reference/config/package-config/). It will be removed after two pack releases with the new feature
+- When `platform.os` is present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/), throw a warning messages indicating the field will be removed 
+and `--target` flag must be used
+- When `platform.os` is not present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/) and `--target` flag is not used, throw a warning messages indicating 
+a new `--target` flag is available
+- Keep doing our current process to package a buildpack
+
+### To improve user experience
+
+We propose:
+- Add a new `--target` flag with the follwing format `[os][/arch][/variant]:[name@version]` to build for a particular target, once the `platform.os` field is removed, 
+this will be the way for end-users to specify the platform for which they want to create single OCI artifact.
+
+- Add `targets` section to `buildpack.toml` to help Buildpack Authors to include support for new platforms without having to update their `pack buildpack package` command in their CI/CD pipelines
+
+- A new folder structure to organize the buildpacks binaries for multi-arch images, similar to this one:
+```bash
+# Option 1 - no variant is required
+.
+├── buildpack.toml                 // mandatory
+└── {os}                           // optional
+    └── {arch}                     // optional
+        └── bin
+            ├── build              // platform dependant binary (mandatory)
+            └── detect             // platform dependant binary (mandatory)
+
+# Option 2 - variant is required
+.
+├── buildpack.toml                  // mandatory
+└── {os}                            // optional
+    └── {arch}                      // optional
+        └── {variant}               // optional
+            ├── {name@version-1}    // optional
+            │ └── bin
+            │     ├── build         // platform dependant binary (mandatory)
+            │     └── detect        // platform dependant binary (mandatory)
+            └── {name@version-2}    // optional
+                └── bin
+                    ├── build       // platform dependant binary (mandatory)
+                    └── detect      // platform dependant binary (mandatory)
+```
+> **Note** 
+> For cross-compile buildpacks like Paketo, it looks easy to add a step to their Makefile to compile and separate the binaries following this structure. It is important to mention
+> that the final buildpack image will not change, this will only change the buildpack structure from `pack` perspective
+
+In case, this folder structure is not suitable for Buildpack Authors, **we propose** a new `path` attribute to be included 
+into the `target` section of the `buildpack.toml` to specify where the binaries are located in the filesystem.
```

Comment:
Would that be the location of the `bin` directory, or the buildpack root directory? I think the latter makes more sense, but I could be wrong. Do we need to add this now, or is this something that could be added later if buildpack authors request it?

Reply:
I was thinking on the **buildpack root directory**, well it came up in a WG meeting suggested by @hone. on the implementation side, I'd propose we could create the ticket and do it later that will simplify the first implementation.